src/geometry/euclidean.lean

```diff
@@ -33,3 +35,634 @@ normed_add_torsor V P
+  ∥x + y∥ * ∥x + y∥ = ∥x∥ * ∥x∥ + ∥y∥ * ∥y∥ ↔ inner x y = 0 :=
+begin
+  rw [norm_add_mul_self, add_right_cancel_iff, add_right_eq_self, mul_eq_zero],
+  norm_num
+end
+
+/-- Pythagorean theorem, subtracting vectors, if-and-only-if vector
+inner product form. -/
+lemma norm_sub_square_eq_norm_square_add_norm_square_iff_inner_eq_zero (x y : V) :
+  ∥x - y∥ * ∥x - y∥ = ∥x∥ * ∥x∥ + ∥y∥ * ∥y∥ ↔ inner x y = 0 :=
+begin
+  rw [norm_sub_mul_self, add_right_cancel_iff, sub_eq_add_neg, add_right_eq_self, neg_eq_zero,
+      mul_eq_zero],
+  norm_num
+end
+
+/-- Pythagorean theorem, if-and-only-if vector angle form. -/
+lemma norm_add_square_eq_norm_square_add_norm_square_iff_angle_eq_pi_div_two (x y : V) :
+  ∥x + y∥ * ∥x + y∥ = ∥x∥ * ∥x∥ + ∥y∥ * ∥y∥ ↔ angle_of_vectors x y = real.pi / 2 :=
+begin
+  rw norm_add_square_eq_norm_square_add_norm_square_iff_inner_eq_zero,
+  exact inner_eq_zero_iff_angle_eq_pi_div_two x y
+end
+
+/-- Pythagorean theorem, subtracting vectors, if-and-only-if vector angle form. -/
+lemma norm_sub_square_eq_norm_square_add_norm_square_iff_angle_eq_pi_div_two (x y : V) :
+  ∥x - y∥ * ∥x - y∥ = ∥x∥ * ∥x∥ + ∥y∥ * ∥y∥ ↔ angle_of_vectors x y = real.pi / 2 :=
+begin
+  rw norm_sub_square_eq_norm_square_add_norm_square_iff_inner_eq_zero,
+  exact inner_eq_zero_iff_angle_eq_pi_div_two x y
+end
+
+/-- Law of cosines (cosine rule), vector angle form. -/
+lemma norm_sub_square_eq_norm_square_add_norm_square_sub_two_mul_norm_mul_norm_mul_cos_angle
+    (x y : V) :
+  ∥x - y∥ * ∥x - y∥ = ∥x∥ * ∥x∥ + ∥y∥ * ∥y∥ - 2 * ∥x∥ * ∥y∥ * (angle_of_vectors x y).cos :=
+by rw [(show 2 * ∥x∥ * ∥y∥ * (angle_of_vectors x y).cos =
+             2 * ((angle_of_vectors x y).cos * (∥x∥ * ∥y∥)), by ring),
+       cos_angle_of_vectors_mul_norm_mul_norm, ←inner_self_eq_norm_square,
+       ←inner_self_eq_norm_square, ←inner_self_eq_norm_square, inner_sub_sub_self,
+       sub_add_eq_add_sub]
+
+/-- Pons asinorum, vector angle form. -/
+lemma angle_sub_eq_angle_sub_rev_of_norm_eq {x y : V} (h : ∥x∥ = ∥y∥) :
+  angle_of_vectors x (x - y) = angle_of_vectors y (y - x) :=
+begin
+  refine real.cos_inj_of_nonneg_of_le_pi (angle_of_vectors_nonneg _ _)
+                                         (angle_of_vectors_le_pi _ _)
+                                         (angle_of_vectors_nonneg _ _)
+                                         (angle_of_vectors_le_pi _ _) _,
+  rw [cos_angle_of_vectors, cos_angle_of_vectors, h, ←neg_sub, norm_neg, neg_sub,
+      inner_sub_right, inner_sub_right, inner_self_eq_norm_square, inner_self_eq_norm_square, h,
+      inner_comm x y]
+end
+
+/-- Converse of pons asinorum, vector angle form. -/
+lemma norm_eq_of_angle_sub_eq_angle_sub_rev_of_angle_ne_pi {x y : V}
+    (h : angle_of_vectors x (x - y) = angle_of_vectors y (y - x))
+    (hpi : angle_of_vectors x y ≠ real.pi) : ∥x∥ = ∥y∥ :=
+begin
+  replace h := real.arccos_inj (abs_le.mp (abs_inner_div_norm_mul_norm_le_one x (x - y))).1
```

Comment:
I would put the hypothesis on a newline, indented by two spaces, and all the others below it, like you did now. This keeps all the line lengths a bit shorter.

Reply:
Adjusted formatting.